### DISCUSSION TITLE
Feed legs naming is confusing (prefer L1/L2/L3 over A/B/C)

### DISCUSSION THREAD

**Discussion by pv2b** [netbox-community/netbox#13344] [Polls]
Hello!

Really getting into Netbox for documenting our datacenter, but I'm running into one piece of confusion in my use case.

At least here in Sweden (I'm not sure if maybe this is different in other countries?), typically the three feed legs of a three-phase power feed will be labeled as L1, L2 and L3, whereas in Netbox they are labeled as A, B and C.

Independently of feed legs, typically the letters A and B will be used to indicate two (at least somewhat) independent power supplies.

As an example, a rack might contain two independent three phase power feeds from two different panels, called A and B. Those feeds would in turn have three feed legs each, L1, L2 and L3.

This is where the potential confusion comes in. Since in Netbox, the feed legs aren't labeled L1, L2 and L3, but instead A, B and C, this makes it very easy to confuse feed legs and independent feeds. (I've found myself getting confused by this at least.)

Before making a proposal to change this though, I'd love to hear from other Netbox users. From my perspective, the solution is to rename the feed legs to L1, L2 and L3. (While also keeping in mind any API or database breakage and compatibility.) But I could be missing some regional context here. Outside of Netbox, does anyone really refer to three-phase feed legs as A, B or C, or is L1, L2 and L3 the more common naming?

(If this turns out to be controversial, then there might be a case for making it configurable, but I'd say that's probably best avoided to avoid complexity.)

**Comment by sleepinggenius2** [comment]
In the US at least, the phases for 3-phase AC power are A, B, and C. For DC power, you typically see A and B feeds. I would certainly have similar confusion if it changed to something other than A, B, and C. It does sound like this is a regional issue, but I'm not sure if there would be a practical way to address it with the internationalization changes being introduced in the 4.0 release or not.

**Reply by jeremystretch** [reply]
We'll definitely have the ability to translate these values, however in this case it seems more like a localization issue than a language one.

**Reply by pv2b** [reply]
Thank you for your input. I was afraid this might be precisely some kind of US:ism I wasn't aware of, and now, knowing it's controversial, there's no use suggesting a change across the board. :-)

Internationalization here would probably be a bit tricky, since the correct way to show this isn't neccessarilly going to be from the point of view of the user, but will instead depend on the region the power feed itself is in. One fairly simple way to deal with this would be to allow the users to define names of their feed legs in the power feed definition itself. This would also be able to handle other power feed scenarios such as split-phase (180 degree) power or more exotic power configurations. (Refer to earlier discussion https://github.com/netbox-community/netbox/discussions/8028, dealing with delta/wye configurations)

**Comment by pv2b** [comment]
Also, I did some more research, and I was able to find this article:
https://www.electricalengineeringtoolbox.com/2015/12/how-to-identify-three-phase-windings.html

I have very serious doubts about how accurate it on the face of it. It suggests that we here in "Europe" (for what it's worth, Europe is a big heterogenous place) use U, V, W or R S T (neither of which I've ever encountered, but I'm not an electrician) and that L1, L2, L3 is a US thing (which seems to be contradicted by both me and @sleepinggenius2). But what it does tell me at least is that there seem to be more than two ways to do it, and giving users a way to label their phases or feed legs to match their naming standard might make more sense than trying to cater to everyone's preference.

**Comment by markkuleinio** [comment]
FWIW, in Finland I have only seen "Feeds = A and B, Phases = 1, 2 and 3", but my actual experience about different datacenter power configurations is limited to only Equinix datacenters.

**Comment by candlerb** [comment]
In one Interxion data centre in the UK, the in-rack PDUs are labelled with "A", "B" or "C". For example, in one rack the left PDU might be "B" and the right one "C"; in another rack they could be "C" and "A" respectively.

We didn't bother to document them though, as we didn't see the need. If one phase goes down then we're fine; and if two phases go down simultaneously, then probably all three will.

**Comment by sleepinggenius2** [comment]
Not to muddy this conversation, but the way that phases are currently used between power outlets and power ports seems to be very similar to the relationship between front ports and rear port positions. I wonder if there might not be a way to implement them in the same way.

One of the things that I've been encountering with rear port positions is that they must be strictly numeric and do not support at least an additional label that could be used, for example, in the case of a fiber cable to indicate something like the fiber and buffer tube colors. This came up recently for me as we started to look at better ways of modeling our outside plant connections in NetBox. I bring this up just to say that a solution to address that could also be workable to address this issue, by allowing users to specify the phase names down to a per-device level, while also allowing for them to be specified at the device type level for general use.

While maybe not intended, we also use the Feed leg (feed_leg) field today to indicate the DC feed for an outlet, which really helps for reporting when we're doing maintenance on our DC plant. We have come across the issue though that in some of our locations we have 4-tier/row distribution, which means we don't have a good way to represent the D feed within our DC distribution, so it would be nice to also allow the flexibility there to support more than 3 feed leg options.

**Comment by De-Backer** [comment]
a Belgian answer, use both e.g.
- A (L1)
- B (L2)
- C (L3)

My personal preference is for L1, L2, L3 because this is used in the EU. (I'm an electrician, U, V, W or R S T is for a motor, generally.)
